### DISCUSSION TITLE
How to pass hostname to nmap, not ips

### DISCUSSION THREAD

**Discussion by Aaronepower** [projectdiscovery/naabu#479] [Q&A]
when I pass naabu to nmap, it will pass ip to nmap not hostname, how can I pass hostname to nmap?

> echo hackerone.com | naabu -silent -nmap-cli 'nmap'

```
hackerone.com:8443
hackerone.com:8080
hackerone.com:443
hackerone.com:80
Starting Nmap 7.80 ( https://nmap.org ) at 2022-11-17 16:15 UTC
Nmap scan report for 104.16.99.52
Host is up (0.00067s latency).

PORT     STATE SERVICE
80/tcp   open  http
443/tcp  open  https
8080/tcp open  http-proxy
8443/tcp open  https-alt

Nmap done: 1 IP address (1 host up) scanned in 0.05 seconds
```

#### Expected:

> echo hackerone.com | naabu -silent -nmap-cli 'nmap'
```
hackerone.com:8443
hackerone.com:8080
hackerone.com:443
hackerone.com:80

Starting Nmap 7.80 ( https://nmap.org ) at 2022-11-17 16:15 UTC
Nmap scan report for hackerone.com
Host is up (0.00067s latency).

PORT     STATE SERVICE
80/tcp   open  http
443/tcp  open  https
8080/tcp open  http-proxy
8443/tcp open  https-alt

Nmap done: 1 IP address (1 host up) scanned in 0.05 seconds
```

**Comment by ehsandeep** [comment]
@Aaronepower @local-host-poc

Could you share the difference in both? if hostname is supported, nmap will again resolve the host to same IP and do the port scan, i.e 1 more additional dns request even we have the IP already.

**Comment by olearycrew** [comment]
@Aaronepower any more information you can share here? If not, we'll consider this discussion closed.

**Comment by dogancanbakir** [comment]
This discussion closed automatically due to inactivity. Feel free to reopen or start new if still relevant.
